### ISSUE TITLE
ASI not working after arguments-less new expression with type arguments

### ISSUE THREAD

**Issue by nicolo-ribaudo** [microsoft/TypeScript#48654]
# Bug Report

### 🔎 Search Terms

asi new expression type arguments

### 🕗 Version & Regression Information

TypeScript 4.7 Nightly (tested with the playground). This syntax has never been valid before TS 4.7.

`new X<T>;` is valid syntax since https://github.com/microsoft/TypeScript/pull/47607 (cc @ahejlsberg)

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.7.0-dev.20220408#code/MYGwhgzhAECCA8AVAfNA3gXwFBYHYFMB3OeXAVwFsAjfAJ2RwEsAzaACgAYBKdDIA)

### 💻 Code

```ts
class A<T> {}

new A<number>

if (0) {}
```

### 🙁 Actual behavior

It's parsed as if it was
```ts
class A<T> {}

new A<number> if (0) {}
```
thus it reports a few errors (`number` is used in a value position, and `if` is a reserved word).

### 🙂 Expected behavior

I would expect ASI to kick, and that cause that code to be parsed as if it was
```ts
class A<T> {}

new A<number>;

if (0) {}
```

**Comment by nicolo-ribaudo** [comment]
Fwiw, the current implementation I'm doing in Babel (unless this is confirmed to be the expected behavior) is to bail out from parsing `A <T>` as an expression with type arguments if it's followed by a non-`(` token that can appear at the beginning of an expression.

**Comment by ahejlsberg** [comment]
Yeah, we're probably being a bit too conservative here. Let me think about it and get back...

**Comment by ahejlsberg** [comment]
I agree we can be a bit more permissive here. I think we should consider an expression of the form `T<X>` to be an expression with type arguments if it is followed by a `(` token, a _NoSubstitutionTemplate_ production, or a _TemplateHead_ production, or any token that can't start an expression. So, basically the rule you suggested, but also allowing for generic tagged template literals.

**Comment by nicolo-ribaudo** [comment]
@ahejlsberg I don't know if it can make sense semantically, but I'm also surprised that this is not parsed as an instantiation expression:
```ts
b<x> ?? 2;
```
